### PR TITLE
[node-forge] Fix `CertificateRequest` interface

### PR DESCRIPTION
**Pull request by Semigradsky**
Closes #66973
Also related: https://github.com/digitalbazaar/forge/issues/1036

Here I moved types to separate files (the same as in the origin repo) for easier future updating. And fixing `CertificateRequest` interface, it was incorrectly extended from `Certificate`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/node-forge/node-forge-tests.ts
Consider running `npx dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 180f088c026566f31764ea50ff6be70deec73bcf
</p>

**Comment by Ichnafi** [comment]
Any ETA when this will be merged? The package is hardly usable the current state.

**Review by andersk — CHANGES_REQUESTED** [review]
This PR is not reviewable in this state because it mixes a sweeping refactor that moves thousands of lines of code with functional changes in the same commit.

Please split this into a commit that makes the functional changes without moving any code, and a commit that moves the code without making any other changes, so that it’s possible to see what you changed by reading the diffs.

**Comment by Semigradsky** [comment]
Ok, no probs

File: types/node-forge/md.d.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        };
+    };
+
+    const md5: typeof md.md5;
+    const sha1: typeof md.sha1;
+    const sha256: typeof md.sha256;
+    const sha512: typeof md.sha512 & {
+        sha384: typeof md.sha512.sha384;
+    };
+    const sha384: typeof md.sha512.sha384;
```
Comment (andersk) [inline types/node-forge/md.d.ts:51]: This is not a pure move.

File: types/node-forge/index.d.ts [hunk 2]
```diff
@@ -8,1539 +35,4 @@ declare module "node-forge" {
...
-            "sha512/224": (typeof md)["sha512/224"];
-            "sha512/256": (typeof md)["sha512/256"];
-        };
-    };
-
-    const md5: typeof md.md5;
-    const sha1: typeof md.sha1;
-    const sha256: typeof md.sha256;
-    const sha384: typeof md.sha384;
-    const sha512: typeof md.sha512;
```
Comment (andersk) [inline types/node-forge/index.d.ts:1147]: (source)

File: types/node-forge/cipher.d.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        interface StartOptions {
+            iv?: util.ByteBuffer | Byte[] | Bytes | undefined;
+            tag?: util.ByteStringBuffer | undefined;
+            tagLength?: number | undefined;
+            additionalData?: string | undefined;
+        }
+
+        interface BlockCipher {
+            start: (options?: StartOptions) => void;
+            update: (input: util.ByteBuffer) => void;
```
Comment (andersk) [inline types/node-forge/cipher.d.ts:27]: This is not a pure move.

File: types/node-forge/index.d.ts [hunk 4]
```diff
@@ -8,1539 +35,4 @@ declare module "node-forge" {
...
-        interface StartOptions {
-            iv?: util.ByteBuffer | Byte[] | Bytes | undefined;
-            tag?: util.ByteStringBuffer | undefined;
-            tagLength?: number | undefined;
-            additionalData?: string | undefined;
-        }
-
-        interface BlockCipher {
-            start: (options?: StartOptions) => void;
-            update: (payload: util.ByteBuffer) => void;
```
Comment (andersk) [inline types/node-forge/index.d.ts:1283]: (source)

File: types/node-forge/cipher.d.ts (hunk 3 above)
Comment (Semigradsky) [reply types/node-forge/cipher.d.ts:27]: It's the same type, just changed parameter name to name from source.

Comment (andersk) [reply types/node-forge/cipher.d.ts:27]: Great—but do it in a separate commit from the pure move.

File: types/node-forge/md.d.ts (hunk 1 above)
Comment (Semigradsky) [reply types/node-forge/md.d.ts:51]: It's this line: https://github.com/digitalbazaar/forge/blob/2bb97afb5058285ef09bcf1d04d6bd6b87cffd58/lib/sha512.js#L23

Comment (andersk) [reply types/node-forge/md.d.ts:51]: I think that was already correctly accounted for by `md.sha384: typeof md.sha512.sha384`, and your added `&` is redundant.

But if there was a problem here to be addressed, do it in a separate commit.

**Review by andersk — CHANGES_REQUESTED** [review]
(review to satisfy bot, see [above](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/66985#pullrequestreview-1700471077))

**Comment by Semigradsky** [comment]
Well, I am not interested in spending so much time on this PR, so let's give someone else the opportunity to do it.

**Pull request closed without merging**